### PR TITLE
Convert new username to lowercase & ignore non-alphanumeric characters

### LINKED ISSUE #4152: Signup: Suggest or correct usernames with capital letters to lowercase
### Expected behavior

If I try to create a username with capital letters in it, I expect the system to autocorrect my chosen username to all lowercase letters. For example, the Calypso signup form at https://wordpress.com/start/account/user autocorrects the username, making it all lowercase if any capital letters were entered.

### Actual behavior

I can enter a username with capital letters, and a message appears to the right:
> Username can only contain lowercase letters (a-z) and numbers

This is problematic for a few reasons:

- It's easy to overlook the word "lowercase" in that message
- It's annoying to have to move the cursor within the field to switch any capital letters to lowercase
- That check runs before the app checks to see if the username already exists, so you can end up correcting the username only to run into the error "That username already exists" when you try to create the account.

### Steps to reproduce the behavior

1. Open the WordPress app (and log out if you are logged in to a WordPress.com account)
2. Select Create Account
3. Enter a new email address, a username with capital letters included, and a password
4. Tap the Create Account button

Result: The "lowercase letters and numbers" message appears

Desired result: After entering a username with capital letters included, the username is autocorrected to all lowercase letters

##### Tested on Nexus 9, Android 6.0.1, WPAndroid alpha-10

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #4152 

To test: After entering a username with capital letters included, the username is auto-corrected to all lowercase letters. Only letters and digits are allowed, all other characters are ignored.

Review: @nbradbury or @theck13 or @maxme or @daniloercoli

### CODE REVIEW

File: WordPress/src/main/res/layout/new_account_user_fragment_screen.xml [hunk 1]
```diff
@@ -87,6 +87,7 @@
                     android:layout_height="wrap_content"
                     style="@style/WordPress.NUXEditText"
                     android:inputType="textUri"
+                    android:digits="0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
```
Comment (daniloercoli) [inline WordPress/src/main/res/layout/new_account_user_fragment_screen.xml]: Did you try with `android:capitalize="none"` and `android:digits="0123456789abcdefghijklmnopqrstuvwxyz"` without using the listener (`afterTextChanged`) that replaces Capital letters?

Comment (0nko) [inline WordPress/src/main/res/layout/new_account_user_fragment_screen.xml]: That would just ignore any capital letters, which might look like a bug to the users. I think it's a better UX to just convert them automatically to lowercase.

Comment (daniloercoli) [inline WordPress/src/main/res/layout/new_account_user_fragment_screen.xml]: Right, I agree with you. 

Noticed that, as reported in a comment in the ticket, we've the same problem in the Blog Address field. One can use capital letters, and punctuation there. Would be good to apply the same fix.

Comment (0nko) [inline WordPress/src/main/res/layout/new_account_user_fragment_screen.xml]: You are right, I missed that. I added the check to the site URL on the new user screen and also the new blog page.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/NewUserFragment.java [hunk 2]
```diff
@@ -397,10 +411,15 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
             @Override
             public void afterTextChanged(Editable s) {
+                String lowerCase = s.toString().toLowerCase();
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/NewUserFragment.java]: Seems that the same code is repeated at `line 391` and in `NewBlogFragment`. Would be possible to create a common method and call it from these 3 different places?

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**